### PR TITLE
cached-thread-pool

### PR DESCRIPTION
**Pull request by DimaGolomozy**

### CODE REVIEW

File: rawhttp-core/src/main/java/rawhttp/core/body/BodyReader.java [hunk 1]
```diff
@@ -113,17 +112,8 @@ public void writeDecodedTo(OutputStream out) throws IOException {
      *                                  by the {@link HttpBodyEncodingRegistry}.
      */
     public void writeDecodedTo(OutputStream out, int bufferSize) throws IOException {
-        DecodingOutputStream decodedStream = framedBody.getBodyDecoder().decoding(out);
-        try {
+        try (DecodingOutputStream decodedStream = framedBody.getBodyDecoder().decoding(out)) {
             framedBody.getBodyConsumer().consumeDataInto(asRawStream(), decodedStream, bufferSize);
-            decodedStream.close();
```
Comment (ophirorly) [inline rawhttp-core/src/main/java/rawhttp/core/body/BodyReader.java]: Why did you remove this section? 
Without it, the caller thread could hang.

**Review by ophirorly — CHANGES_REQUESTED** [review]

File: rawhttp-core/src/main/java/rawhttp/core/body/BodyReader.java (hunk 1 above)
Comment (DimaGolomozy) [reply rawhttp-core/src/main/java/rawhttp/core/body/BodyReader.java]: its still there. i changed it to java `try-with-resources Statement`:
```
try (DecodingOutputStream decodedStream = framedBody.getBodyDecoder().decoding(out))
```
will close the `decodedStream` once it exits the "try" section

**Review by adamwkaczmarek — APPROVED** [review]

File: rawhttp-core/src/main/java/rawhttp/core/body/BodyReader.java (hunk 1 above)
Comment (ophirorly) [reply rawhttp-core/src/main/java/rawhttp/core/body/BodyReader.java]: Oh right! I strayed away from Java for too long 😅

**Review by ophirorly — APPROVED** [review]
